src/node_buffer.cc

```diff
@@ -1004,7 +1004,7 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
   SPREAD_BUFFER_ARG(args[0], ts_obj);
 
   uint32_t needle = args[1]->Uint32Value();
-  int64_t offset_i64 = args[2]->IntegerValue();
+  int64_t offset_i64 = args[2].As<Integer>()->Value();
```

Comment:
As far as I can tell, the JS wrapper method only coerces to a *number* with `+`, not to an integer.